exp/views/video.py

```diff
@@ -28,43 +33,52 @@ def post(self, request):
         # of events in the future; some of the Pipe event data includes semicolons which
         # throw the automatic parsing for a loop.
 
-        # Authenticate the webhook (see https://addpipe.com/docs#authenticating-webhooks)
         key = settings.PIPE_WEBHOOK_KEY
-        # Append the JSON POST data received via webhook to the URL string.
-        thisURL = request.scheme + "://" + request.headers["host"] + request.path
-        payload = request.POST.get("payload")
-        message = thisURL + payload
-        # Hash the resulting string with HMAC-SHA1, using the webhook authentication key; generate binary signature.
-        keyBytes = bytes(key, "UTF-8")
-        messageBytes = bytes(message, "UTF-8")
-        digester = hmac.new(keyBytes, messageBytes, hashlib.sha1)
-        signatureBinary = digester.digest()
-        # Base64 encode the binary signature.
-        signatureComputed = base64.b64encode(signatureBinary)
-        # Compare to the one in the header
-        signatureSent = bytes(request.headers["x-pipe-signature"], "UTF-8")
-        authenticated = signatureComputed == signatureSent
 
-        # Convert data from string representation of dict to dict
-        payload_data = json.loads(payload)
+        try:
+            # Authenticate the webhook (see https://addpipe.com/docs#authenticating-webhooks)
+            # Append the JSON POST data received via webhook to the URL string.
+            thisURL = request.scheme + "://" + request.headers["host"] + request.path
```

Comment:
@okaycj should this URL construction be updated to use the methods you're using in #1641? If so we can merge that first and I'll update it here.

Reply:
Maybe you could use a simpler version of [this](https://stackoverflow.com/a/15799706).